env/work/srw_python/uti_math.py

```diff
@@ -612,12 +612,23 @@ def trf_rotation(_V, _ang, _P):
     return [M, V]
 
 #****************************************************************************
-def fwhm(x, y): #MR27092016
-    """The function searches x-values (roots) where y=0 based on linear interpolation, and calculates FWHM"""
+def fwhm(x, y, shift=0.5, return_as_dict=False):  # MR21032017
+    """The function searches x-values (roots) where y=0 (after normalization to values between 0 and 1 and shifting the
+    values down by 0.5 (default value)) based on linear interpolation, and calculates full width at half maximum (FWHM).
+
+    :param x: an array of x values.
+    :param y: an array of y values.
+    :param shift: an optional shift to be used in the process of normalization (between 0 and 1).
+    :param return_as_dict: if to return a dict with 'fwhm' and 'x_range'
+    :return: a value of the FWHM or dictionary consisting of 'fwhm' and 'x_range'
+    """
 
     def is_positive(num):
         return True if num > 0 else False
 
+    # Normalize values first:
+    y = (y - min(y)) / (max(y) - min(y)) - shift  # roots are at Y=0
```

Comment:
Maksim, this doesn't seem to work with Python native array.array()...
This is what I get when trying to execute SRWLIB_Example15.py:
Traceback (most recent call last):
  File "C:\SoftwareDevelopments\SRW_Dev\env\work\srw_python\SRWLIB_Example15.py", line 191, in <module>
    xRMS.append(fwhm(x, arIxh))
  File "C:\SoftwareDevelopments\SRW_Dev\env\work\srw_python\uti_math.py", line 636, in fwhm
    y = (y - min(y)) / (max(y) - min(y)) - shift  # roots are at Y=0
TypeError: unsupported operand type(s) for -: 'list' and 'float'

Reply:
If you are busy, I'll fix this myself. I want to update SRW on GitHub today or tomorrow.

Reply:
Fixed in 0aa2404.